src/figuresClasses.ts

```diff
@@ -1,11 +1,98 @@
-export interface Figure {}
+export enum Color {
+  Red = 'red',
+  Blue = 'blue',
+  Green = 'green',
+}
+
+export enum Shape {
+  Triangle = 'triangle',
+  Rectangle = 'rectangle',
+  Circle = 'circle',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be positive numbers');
+    }
+
+    if (!Triangle.isValidTriangle(a, b, c)) {
+      throw new Error("These sides can't create a triangle");
+    }
+
+    this.color = color;
+  }
+
+  static isValidTriangle(side1: number, side2: number, side3: number): boolean {
+    const maxSide = Math.max(side1, side2, side3);
+    const sumOfOtherSides = side1 + side2 + side3 - maxSide;
 
-export class Triangle implements Figure {}
+    return maxSide < sumOfOtherSides;
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      halfPerimeter *
+        (halfPerimeter - this.a) *
+        (halfPerimeter - this.b) *
+        (halfPerimeter - this.c),
+    );
 
-export class Rectangle implements Figure {}
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be a positive numbers');
+    }
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
```

Comment:
[Not critical] Handling of area (`Math.floor(area * 100) / 10`) occurs in multiple places, so it makes sense to define it in a separate method, in order to avoid self-duplicating.